units/XNA0305/XNA0305_script.lua

```diff
@@ -53,31 +47,17 @@ XNA0305 = Class(NAirUnit) {
     ---@param groundWeapons boolean
     ---@param airWeapons boolean
     EnableWeapons = function(self, enable, delay, groundWeapons, airWeapons)
-
         if self.DisableWeaponsThread then
             KillThread( self.DisableWeaponsThread )
             self.DisableWeaponsThread = nil
         end
 
-        local fn = function(self, disable, delay, groundWeapons, airWeapons)
-            if delay and delay > 0 then
-                WaitSeconds( delay )
-            end
-            if groundWeapons or true then
-                self:SetWeaponEnabledByLabel( 'CannonLeft', enable or true )
-                self:SetWeaponEnabledByLabel( 'CannonRight', enable or true )
-            end
-            if airWeapons or true then
-                self:SetWeaponEnabledByLabel( 'AAMissile', enable or true )
-            end
-            self.DisableWeaponsThread = nil
-        end
-        self.DisableWeaponsThread = self:ForkThread( (enable == true), delay, groundWeapons, airWeapons )
+        self.DisableWeaponsThread = self.Trash:Add(ForkThread( (enable == true), delay, groundWeapons, airWeapons,self ))
```

Comment:
I think `fn` is the function that's supposed to be used for the thread here. I wouldn't remove it as if this is a refactor, since it being unused might be a bug.
There is no git history to decide if this was broken on accident.